examples/string_token_substitution.ipynb

```diff
@@ -0,0 +1,247 @@
+    "* Hosts/managers need to know which tokens are possible/allowed within a given string.\n",
+    "* Hosts/managers need the ability to substitute values into the placeholder tokens in a string.\n",
+    "\n",
+    "OpenAssetIO solves these problems by a combination of documentation and a handy `substitute` utility function. The documentation of individual traits and their properties will tell hosts/managers whether they can place substitution tokens in a string, what tokens are allowed, and what values to map to each token. A common OpenAssetIO-dictated format for substitution tokens ensures that hosts/managers know how placeholders should be presented within strings. Once the host/manager has gathered the necessary values, they can use the `substitute` utility function to perform the substitution."
+   ],
+   "id": "feec93d227a1e95e"
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "\n",
+    "## Example - LocatableContent\n",
+    "\n",
+    "Let's take an example from the MediaCreation library. The LocatableContent trait has an `\"isTemplated\"` property, which, if true, indicates that the `\"location\"` property contains substitution tokens.\n",
+    "\n",
+    "The `\"isTemplated\"` documentation says\n",
+    "\n",
+    "> The URL in the location property contains variables (post decoding) that must be expanded before any loading is attempted.\n",
+    ">\n",
+    "> Variables use the OpenAssetIO syntax (eg: \"image.{frame:04d}.{view}.exr\") see the OpenAssetIO documentation for more details.\n",
+    ">\n",
+    "> The following well-known variables are defined within the MediaCreation ecosystem, and must be used where applicable to any specific workflow:\n",
+    "> - frame: An integer frame number for the current time.\n",
+    "> - view: A string representing the current view (eg. \"left\").\n",
+    "\n",
+    "This implies that a file name of `\"image.{frame}.{view}.exr\"` should be resolved to `\"image.1.left.exr\"` if the frame number is 1 and the view is \"left\".\n",
+    "\n",
+    "\n"
+   ],
+   "id": "e1cf4ef1d84c6d3c"
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "## The OpenAssetIO substitution syntax\n",
+    "\n",
+    "In general, the substitution tokens are of the form `{variable:format}` where `variable` is the name of the variable and `format` is a format string. The `format` part is optional. If included, the format specification is compatible with the C++ [libfmt](https://fmt.dev/latest/syntax.html) library, which in turn is broadly compatible with C++20's [`std::format`](https://en.cppreference.com/w/cpp/utility/format/format) and Python [format strings](https://docs.python.org/3.9/library/string.html#formatspec). \n",
+    "\n",
+    "For maximum cross language compatibility, only basic substitutions are officially supported. Specifically, substitutions with no format string, or integers with zero padding. This may be revised in the future.\n"
```

Comment:
[Fix] This is confusing. I don't understand if I'm allowed to use format strings, (ie, the second part of the {variable:format} bit mentioned above). We seem to say that substitution tokens allow format strings, and then say substitutions with no format string are the only substitutions supported.

Is what we're saying here really that you may only use a format string if the variable is an integer with zero padding? Quite confused on this. Examples may help.